### PR TITLE
Feature/[LJI-43] implement package contribution.

### PR DESCRIPTION
**Pull request by HotoMazur**
Add the checker to the package contribution. If add a new plan and assign it to a specific package, it checks if we have a contribution for this plan or not.

### CODE REVIEW

**Comment by SavaLX** [comment]
Run pipeline please for your PR

File: src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java [hunk 1]
```diff
@@ -15,4 +17,6 @@ public interface PlanRepository extends JpaRepository<PlanEntity, UUID>,
   boolean existsById(UUID id);
 
   Optional<PlanEntity> findById(UUID id);
+
+  List<PlanEntity> findAllByPackageId(PackageEntity packageEntity);
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java:21]: why are you passing the entire entity?

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java [hunk 2]
```diff
@@ -98,4 +105,18 @@ private boolean isNotEmpty(String value) {
     return value != null && !value.isEmpty();
   }
 
+  private void validateContributionLimit(PlanEntity planEntity) {
+    double totalContributions =
+        planRepository.findAllByPackageId(planEntity.getPackageId()).stream()
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java:110]: find the sum through the SQL and don't fetch all entities

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java [hunk 3]
```diff
@@ -98,4 +105,18 @@ private boolean isNotEmpty(String value) {
...
+            .mapToDouble(PlanEntity::getTotalLimit)
+            .sum();
+    PackageEntity packageContribution = packageRepository.findById(planEntity.getPackageId().getId())
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java:113]: The misleading name of the variable

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java [hunk 4]
```diff
@@ -98,4 +105,18 @@ private boolean isNotEmpty(String value) {
     return value != null && !value.isEmpty();
   }
 
+  private void validateContributionLimit(PlanEntity planEntity) {
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java:108]: Add negative and positive unit tests for this logic

**Review by SavaLX — CHANGES_REQUESTED** [review]

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java (hunk 4 above)
Comment (HotoMazur) [reply src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java:108]: Done

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java (hunk 3 above)
Comment (HotoMazur) [reply src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java]: Done

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java (hunk 2 above)
Comment (HotoMazur) [reply src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java]: Done

File: src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java (hunk 1 above)
Comment (HotoMazur) [reply src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java]: Done

**Comment by HotoMazur** [comment]
@SavaLX, I do not understand, how I can run pipeline by myself. How I understand that we have the triggers in github/workflofs which should start run pipeline, when we create or update pr. So maybe you have some advice?

**Comment by SavaLX** [comment]
@You have to merge the main branch into your branch

File: src/main/java/com/coherentsolutions/pot/insurance/entity/PlanEntity.java [hunk 5]
```diff
@@ -39,4 +41,10 @@ public class PlanEntity extends BaseEntity {
 
   @Enumerated(EnumType.STRING)
   private PlanStatus status;
+
+  @ManyToOne
+  @JoinColumn(name = "package_id", referencedColumnName = "id")
+  private PackageEntity packageId;
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/entity/PlanEntity.java:47]: The property type is 'PackageEntity', but you named the property packageId. It is a misleading name

File: src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java [hunk 6]
```diff
@@ -15,4 +17,7 @@ public interface PlanRepository extends JpaRepository<PlanEntity, UUID>,
   boolean existsById(UUID id);
 
   Optional<PlanEntity> findById(UUID id);
+
+  @Query("SELECT COALESCE(SUM(p.totalLimit), 0) FROM PlanEntity p WHERE p.packageId.id = :packageId")
+  double findSumOfTotalLimitByPackageId(@Param("packageId") UUID packageId);
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java:22]: No need to add @Param

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java [hunk 7]
```diff
@@ -98,4 +105,16 @@ private boolean isNotEmpty(String value) {
...
+    double totalContributions =
+        planRepository.findSumOfTotalLimitByPackageId(planEntity.getPackageId().getId());
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java:110]: `planEntity.getPackageId().getId()` extract to the local variable

**Review by SavaLX — CHANGES_REQUESTED** [review]

File: src/main/java/com/coherentsolutions/pot/insurance/entity/PlanEntity.java (hunk 5 above)
Comment (HotoMazur) [reply src/main/java/com/coherentsolutions/pot/insurance/entity/PlanEntity.java]: Done. Rename to packageEntity

File: src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java (hunk 6 above)
Comment (HotoMazur) [reply src/main/java/com/coherentsolutions/pot/insurance/repository/PlanRepository.java]: Delete

File: src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java (hunk 7 above)
Comment (HotoMazur) [reply src/main/java/com/coherentsolutions/pot/insurance/service/PlanService.java]: Create local variable packageId

**Review by anton-kolyaev — APPROVED** [review]

**Review by NormantasN — APPROVED** [review]

**Review by SavaLX — APPROVED** [review]

**Pull request merged**